src/sbvr-api/sbvr-utils.ts

```diff
@@ -1094,7 +1097,7 @@ const $getAffectedIds = async ({
 	tx,
 }: HookArgs & {
 	tx: Db.Tx;
-}): Promise<number[]> => {
```

Comment:
Why did you change this?

Reply:
According to the odata spec, id should be a string, not a number. See: https://docs.oasis-open.org/odata/odata-json-format/v4.01/odata-json-format-v4.01.html#_Toc38457781:~:text=The%20value%20of%20id%20is%20a%20string%20containing%20the%20request%20identifier%20of%20the%20individual%20request

That is why I changed the typing for anything related to `id`s from `number` to `string`. Aside from just being a specified typing from the spec, it also enables us to use ids such as `"device"` to identify i.e. a device request and then refer to it in a subsequent request such as via `dependsOn: ["device"]`.

This is also the explanation for the following comment: https://github.com/balena-io/pinejs/pull/675/files/b5f6e0e7b5428c2642790f78c748a70d68998ea9#r1265570856

Reply:
@fisehara So even though the spec specifies that the property is called `id`, I need to name it something else (i.e. `requestId`) in pinejs?